pkg/sql/sessiondata/session_data.go

```diff
@@ -236,7 +236,11 @@ func (s *Stack) Push(elem *SessionData) {
 
 // PushTopClone pushes a copy of the top element to the stack.
 func (s *Stack) PushTopClone() {
-	s.Push(s.Top().Clone())
+	if len(s.stack) == 0 {
+		return
+	}
+	sd := s.stack[len(s.stack)-1]
```

Comment:
this needs to be a clone; otherwise we are doing a shallow copy and uh yeah that ruins SET LOCAL

Reply:
whoops i committed this by accident. this was from when i was playing around with adding locks to all these functions. had to be inlined so `Lock()` couldn't get called twice